pkg/commands/compute/language_go.go

```diff
@@ -125,14 +160,74 @@ func (g *Go) Build() error {
 	return bt.Build()
 }
 
+// identifyTinyGoConstraint checks the compute-sdk-go version used by the
+// project and if it's less than 0.2.0 we'll change the TinyGo constraint to be
+// version 0.26.0
+//
+// We do this because the 0.2.0 release of the compute-sdk-go bumps the TinyGo
+// version requirement to 0.28.1 and we want to avoid any scenarios where a
+// bump in SDK version causes the user's build to break (which would happen for
+// users with a pre-existing project who happen to update their CLI version: the
+// new CLI version would have a TinyGo constraint that would be higher than
+// before and would stop their build from working).
+//
+// NOTE: The `configConstraint` is the latest CLI application config version.
+// If there are any errors trying to parse the go.sum we'll default to the
+// config constraint.
+func identifyTinyGoConstraint(configConstraint string) string {
+	moduleName := "github.com/fastly/compute-sdk-go"
+	version := ""
+
+	f, err := os.Open("go.sum")
```

Comment:
Why examine `go.sum` rather than `go.mod`?  I think the module file would be a more reliable indicator of the version the user wanted.

There is a parser for `go.mod` files available in https://pkg.go.dev/golang.org/x/mod/modfile.

`go.sum` can contain multiple versions, so if you stick with `go.sum` you'll need to handle that case.

Reply:
I'll switch to `go.mod`

I opted for `go.sum` originally as I thought it would be easier to parse (as go.mod had different ways of defining a dependency) but I neglected to realise that the dependency could appear multiple times.